Upgrade to python3.13 and other dependencies

lumpy72006 opened the pull request:
fix #132 
Ready for review

benoit74 commented:
- is this ready for review? (please inform us everytime PR is ready for review, or it is impossible for us to know if it is still ongoing work or complete)
- why did you had to add `pytest` to `check` dependencies? Unless I'm mistaken, it is not supposed to be used by `check` tasks and is already part of the `test` dependencies?
- please add `Fix #xxx` text in first comment to link PR with issue (where xxx is fixed issue number ; multiple issues might be fixed by one PR): https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue

lumpy72006 commented:
> * is this ready for review? (please inform us everytime PR is ready for review, or it is impossible for us to know if it is still ongoing work or complete)
> * why did you had to add `pytest` to `check` dependencies? Unless I'm mistaken, it is not supposed to be used by `check` tasks and is already part of the `test` dependencies?
> * please add `Fix #xxx` text in first comment to link PR with issue (where xxx is fixed issue number ; multiple issues might be fixed by one PR): https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue

I had to add it because when I ran `hatch run check:all`, I got a total of 205 errors and this  `/home/lumpy/mindtouch/scraper/tests/test_asset.py:1:8 - error: Import "pytest" could not be resolved (reportMissingImports)` was the first. I thought maybe `check` couldn't access `pytest` and when I added it I got 0 errors.

lumpy72006 commented on scraper/pyproject.toml:
Turns out adding `pytest` to `check` was unnecessary. :sweat_smile:

lumpy72006 commented:
> > * is this ready for review? (please inform us everytime PR is ready for review, or it is impossible for us to know if it is still ongoing work or complete)
> > * why did you had to add `pytest` to `check` dependencies? Unless I'm mistaken, it is not supposed to be used by `check` tasks and is already part of the `test` dependencies?
> > * please add `Fix #xxx` text in first comment to link PR with issue (where xxx is fixed issue number ; multiple issues might be fixed by one PR): https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue
> 
> I had to add it because when I ran `hatch run check:all`, I got a total of 205 errors and this `/home/lumpy/mindtouch/scraper/tests/test_asset.py:1:8 - error: Import "pytest" could not be resolved (reportMissingImports)` was the first. I thought maybe `check` couldn't access `pytest` and when I added it I got 0 errors.


> > * is this ready for review? (please inform us everytime PR is ready for review, or it is impossible for us to know if it is still ongoing work or complete)
> > * why did you had to add `pytest` to `check` dependencies? Unless I'm mistaken, it is not supposed to be used by `check` tasks and is already part of the `test` dependencies?
> > * please add `Fix #xxx` text in first comment to link PR with issue (where xxx is fixed issue number ; multiple issues might be fixed by one PR): https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue
> 
> I had to add it because when I ran `hatch run check:all`, I got a total of 205 errors and this `/home/lumpy/mindtouch/scraper/tests/test_asset.py:1:8 - error: Import "pytest" could not be resolved (reportMissingImports)` was the first. I thought maybe `check` couldn't access `pytest` and when I added it I got 0 errors.

lumpy72006 commented:
I fixed it now. Sorry if I'm stressing you :pray:

benoit74 reviewed (changes requested):
See inline comments for small changes + CI is failing, the `Dockerfile` base image needs to be updated to use Python 3.13 as well (it wasn't specified in the issue tbh).
